keyboards/monsgeek/m1/config.h

```diff
@@ -23,6 +23,9 @@
 /* Force NKRO on boot up regardless of the setting saved in the EEPROM (uncomment to enable it) */
 #define FORCE_NKRO
 
+/* encoder */
+#define ENCODER_MAP_KEY_DELAY 20
+
```

Comment:
When it is set to 10, the volume may jump twice when the knob is turned at a very slow speed. I think if I use the encoder to fine-tune the value and only turn it once but report twice, it would be more unpleasant.But setting it to 20 will cause the problem you described.
The experience of using the knob is ultimately subjective.So I commented out the settings related to the encoder and restored them to the default values.
You can still open them again and set them to the best value you think.
 ` #define ENCODER_RESOLUTION 4 ` #define` ENCODER_MAP_KEY_DELAY 10``

Reply:
For my testing I already removed the encoder resolution from info.json and added to config.h.

If the encoder Resoultion is changed in the info.json or config.h gives the same result. I don not know which way is preferred in QMK.

Encoder Resolution set to 2 gives me similar result as the default Monsgeek VIA V3 Firmware.
The encoder as haptical feedfack.
Between each "click" of the encoder it seems to send 4 pulses.
- with setting the encoder_resolution to 4 only 1 pulse from this gets registered and reported to the operating system
- with encoder_resolution at 2, 2 pulses are registered
- with encoder_resolutiuon at 1, all 4 pulses are registered

And with fine manual turning of the knob i can set all increments between the clicks of the encoder.
Like it is said in the documentation:
_Additionally, the resolution, which defines how many pulses the encoder registers between each detent, can be defined with:_
`#define ENCODER_RESOLUTION 4`
(https://github.com/qmk/qmk_firmware/blob/master/docs/feature_encoders.md)

Regarding the ENCODER_MAP_KEY_DELAY, https://github.com/qmk/qmk_firmware/blob/master/docs/feature_encoders.md:
_Using encoder mapping pumps events through the normal QMK keycode processing pipeline, resulting in a keydown/keyup combination pushed through process_record_xxxxx(). To configure the amount of time between the encoder "keyup" and "keydown", you can add the following to your config.h:_
`#define ENCODER_MAP_KEY_DELAY 10`
_By default, the encoder map delay matches the value of TAP_CODE_DELAY._

Default TAP_CODE_DELAY should be 0, https://github.com/qmk/qmk_firmware/blob/master/docs/config_options.md:
`#define TAP_CODE_DELAY 100`
_Sets the delay between register_code and unregister_code, if you're having issues with it registering properly (common on VUSB boards). The value is in milliseconds and defaults to 0._

As far as I understand ENCODER_MAP_KEY_DELAY has the same functionality as TAP_CODE_DELAY. But only affects encoder signals, where TAP_CODE_DELAY affects every key.


This is how the additional settings in config.h are for me at the moment and they seem to work fine:
```
/* encoder */
//#define ENCODER_MAP_KEY_DELAY 20
#define ENCODER_RESOLUTION 2

/* Change USB Polling Rate to 1000hz */
#define USB_POLLING_INTERVAL_MS 1
```

Reply:
If you want the have in the default firmware 1 increment per 1 "click" of the encoder this should be the setting:
`#define ENCODER_RESOLUTION 4`
ENCODER_MAP_KEY_DELAY should only be used if there are further problems, i think.

Reply:
> /* Change USB Polling Rate to 1000hz */

`USB_POLLING_INTERVAL_MS` is already 1ms ... see `tmk_core/protocol/usb_descriptor.c` or `tmk_core/protocol/vusb/vusb.c` files, or see PR #15352, posted back in Nov 2021.

Reply:
I later have seen that the polling rate default is already 1kHz.

Regarding the encoder:
I have a Monsgeek M1, and tested it with volume in Windows and slow turning the knob.
With `#define ENCODER_RESOLUTION 1` i get 4 steps from one "click" to the next
With `#define ENCODER_RESOLUTION 2` i get 2 steps from one "click" to the next
With `#define ENCODER_RESOLUTION 4` i get 1 step from one "click" to the next
I do not know which encoder Monsgeek M1 has, but i see the results.